src/sdg/api/authentication.py

```diff
@@ -1,21 +1,14 @@
-from django.contrib.auth.models import AnonymousUser
-
+from rest_framework import exceptions
 from rest_framework.authentication import TokenAuthentication as _TokenAuthentication
 
-from sdg.api.models import Token
+from .models import TokenAuthorization
 
 
 class TokenAuthentication(_TokenAuthentication):
-    """Custom token authentication without user binding."""
-
-    model = Token
-
     def authenticate_credentials(self, key):
-        """Authenticate credentials without user checking."""
-        anon, model = AnonymousUser(), self.get_model()
-
         try:
-            token = model.objects.get(key=key)
-            return anon, token
-        except model.DoesNotExist:
-            return anon, None  # optional
+            token = TokenAuthorization.objects.get(token=key)
+        except TokenAuthorization.DoesNotExist:
+            raise exceptions.AuthenticationFailed("Invalid token.")
+
+        return (token.lokale_overheid, token)
```

Comment:
This seems wrong. You link the authorization to a specific lokale overheid but the idea is that with 1 token you can manage multiple lokale overheden.

This line `TokenAuthorization.objects.get(token=key)` is also a bit nasty because what you are actually saying is: `TokenAuthorization.objects.get(token__key=key)`.

Anyway, the old way is better.